bond_app/authentication.py

```diff
@@ -74,17 +77,21 @@ def require_user_info(self, request_state, token_info_fn=_token_info):
 
         token = auth_header_parts[1]
 
-        user_info = self.cache_api.get(key='access_token:' + token)
-        if user_info is None:
-            user_info = self._fetch_user_info(token, token_info_fn)
+        google_user_info = self.cache_api.get(key='access_token:' + token)
+        if google_user_info is None:
+            google_user_info = self._fetch_user_info(token, token_info_fn)
             # cache for 10 minutes or until token expires
-            expires_in = min([user_info.expires_in, self.config.max_token_life])
+            expires_in = min([google_user_info.expires_in, self.config.max_token_life])
             logging.debug("caching token %s for %s seconds", token, expires_in)
-            self.cache_api.add(key='access_token:' + token, value=user_info, expires_in=expires_in)
+            self.cache_api.add(key='access_token:' + token, value=google_user_info, expires_in=expires_in)
         else:
             logging.debug("auth token cache hit for token %s", token)
 
-        return user_info
+        sam_user_info = self.sam_api.user_info(google_user_info)
+        if sam_user_info is None or not sam_user_info[SamKeys.USER_ENABLED_KEY]:
+            raise exceptions.Unauthorized("user not found in sam")
```

Comment:
I can appreciate having this be the error raised, in the spirit of not returning a message that indicates the user exists but is just not enabled.

That said, we may benefit from a log statement here to show the response from Sam, might be helpful in debugging?

Reply:
Yeah, I wasn't quite sure what to do with this error message either. I left it in the interest of not returning too informative of a message, but I kind of want to change it to something along the lines of "user is not registered in Sam" or even just removing the message entirely and just returning the 401 default message

Reply:
With added logging though, so _we_ can at least see what's happening

Reply:
Right, in terms of the message returned, I would want to return the most descriptive and precise message we can that is reasonable and secure.  Maybe sec ops has an opinion about whether we can say "user not enabled in Sam".  

At the very least, get that information into the logs so that when a user does have a problem and they go, "But I'm registered!" we can look in the logs and get the details.

Reply:
How do you like the message I changed it to, "could not authenticate with Sam" along with a log message that includes the user's email, google subject id, and the response from Sam?